### PR TITLE
[styles] Add castle_types

### PR DESCRIPTION
**Pull request by j13m126**
This PR adds all remaining castle types listed in [wiki](https://wiki.openstreetmap.org/wiki/Key%3Acastle_type), 
namely palace, manor, fortress, castrum, shiro, kremlin, hillfort and fortified_church.
A few points to discuss here:

- castrum, shiro, kremlin, hillfort and fortified_church are all below 101 occurrences each but I didn't want to leave them out since they are culturally and historically specific. Shiro could be dropped, I think, since they are all in Japan (except 2 in s.korea) and translations are all 'japanese castle' (including japanese).
 
-  should all castle_types be searchable? or generally as castle? imo one category should be enough.

- I replaced [historic=fort](https://wiki.openstreetmap.org/wiki/Tag%3Ahistoric%3Dfort) with [castle_type=fortress](https://wiki.openstreetmap.org/wiki/Tag:castle_type%3Dfortress) and [historic=manor](https://wiki.openstreetmap.org/wiki/Tag%3Ahistoric%3Dmanor) with [castle_type=manor](https://wiki.openstreetmap.org/wiki/Tag:castle_type%3Dmanor) in `replaced_tags.txt`as most translations don't differ and they're pretty similar.

- I added all tags to `mapcss-mapping.csv` like the existing castle types defensive and stately, but they're not showing up yet after map generation. Any ideas why? They are all still labeled 'stately'.

- I removed the brown background from remains.svg (no business). Do you want to keep the grey color? Right now all historic and culturally associated icons have a brown background and I think it's a nice highlight of historic sights. OSM also colors castles and culturally associated sights brown.
   - the icons used rn for tag historic are: tomb, wayside_shrine, monument, remains & plaque (rn grey withough background).

I can change all four icons for consistency if you want.
Here is a little preview of what it would look like (plaque, remains & monument.svg in grey vs brown):
Bildschirmfoto vom 2022-08-05 10-59-13

Bildschirmfoto vom 2022-08-05 11-30-37

I will merge all commits when everything is clear.

### CODE REVIEW

**Review by vng — APPROVED** [review]
Icons background is debated. Classifier approach is ok.

**Review by biodranik — COMMENTED** [review]
The brown color is great, it is used on many real signs for sightseeing purposes in many countries.

Background... Highlighting touristic features is a good idea, but how to distinguish them from other businesses?

**Comment by biodranik** [comment]
Can you please also add a historic=city_gate, as a bonus? :)

File: data/replaced_tags.txt [hunk 1]
```diff
@@ -101,6 +101,8 @@ historic=cannon : historic=memorial
 historic=technical_monument : historic=memorial
 historic=shield : historic=memorial
 historic=war_memorial : historic=memorial
+historic=manor : castle_type=manor
+historic=fort : castle_type=fortress
```
Comment (vng) [inline data/replaced_tags.txt:105]: Not sure about this. This kind of replacement will not match tags into historic-castle-manor/fortress.

**Comment by vng** [comment]
@j13m126 If you finished your idea here with icons and styles, I can take this PR and finalize it with fixing C++ tests and replaced_tags. If you don't mind, of course.

**Comment by biodranik** [comment]
Cool! Is it possible to make tombs smaller? They look too big on screenshots and attract too much attention.

**Comment by biodranik** [comment]
Also, for the night theme, the brown color doesn't look nice. Maybe change it to some gray, like the information board on one of the screenshots?

**Comment by vng** [comment]
@j13m126 Ok, feel free to update svg icons or colors/styles. I'll do the rest ;)

**Comment by vng** [comment]
@j13m126 I'd like to make force update of your branch. Are you ok with that?

**Comment by vng** [comment]
I kept historic=fort as a separate type according to the OSM wiki.

**Comment by vng** [comment]
Good, at least better than before. Could you please rebase, regenerate styles/symbols/strings and we will merge it.

**Pull request merged**